cmd/stack_lint_test.go

```diff
@@ -286,44 +287,327 @@ func TestLintWithValidStack(t *testing.T) {
 	}
 }
 
-func TestLintWithMissingConfig(t *testing.T) {
+func TestLintWithMissingStackYaml(t *testing.T) {
 	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
 	defer cleanup()
 
-	currentDir, _ := os.Getwd()
-	testStackPath := filepath.Join(currentDir, "testdata", "test-stack")
+	testStackPath := filepath.Join(cmdtest.TestDirPath, "test-stack")
 	args := []string{"stack", "lint", testStackPath}
-	_, err := cmdtest.RunAppsody(sandbox, args...)
-
-	if err != nil { //Lint check should pass, if not fail the test
+	removeYaml := filepath.Join(testStackPath, "stack.yaml")
+	file, err := ioutil.ReadFile(removeYaml)
+	if err != nil {
 		t.Fatal(err)
 	}
+	removeArray := []string{filepath.Join(removeYaml)}
+
+	osErr := os.RemoveAll(removeYaml)
+	if osErr != nil {
+		t.Fatal(osErr)
+	}
+
+	_, appsodyErr := cmdtest.RunAppsody(sandbox, args...)
+
+	if appsodyErr == nil { //Lint check should fail, if not fail the test
+		t.Fatal("Expected failure - Missing stack.yaml")
+	}
+
+	RestoreSampleStack(removeArray, testStackPath)
+	writeErr := ioutil.WriteFile(removeYaml, []byte(file), 0644)
+	if writeErr != nil {
+		t.Fatal(writeErr)
+	}
```

Comment:
Could you add the `RestoreSampleStack` into a defer (or make the current defer into it a defer function which calls it alongside `cleanup`) so if the test is quit during the run, the files are still restored?

It should probably be done for all other instances too.

Reply:
Apart from this, the rest looks good to me!

Reply:
Good idea - Will add this in the next commit.